### PR TITLE
Replace "unable to" with "failed to" in error message

### LINKED ISSUE #256: Unify error message language for failed allocations
I noticed that the error messages for when `calloc` returns `NULL` use different strings in different parts of the codebase.

To see an example of what I'm talking about, try this against the root of the repo:
`for s in "Unable" "Failed" "Allocation"; do grep -rn -A3 calloc | grep $s; done`

There are probably other contexts where a similar thing happens. I think the most thorough fix for this would probably look like enumerating those contexts and perhaps adding some standards to the coding style guide.

### PR DESCRIPTION
**Pull request by haoruan**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
The error message format is inconsistent throughout the codebase, with a mix of 'Unable to xxx' and 'Failed to xxxx' formats. Since the 'Failed to' format is more commonly used in the codebase, this PR replaces all instances of 'Unable to' with 'Failed to' to unify the error message string.

Also add the guideline for contributing.

#### Which issue(s) this PR fixes:
Fixes #256 

Signed-off-by: Hao Ruan <hao.ruan@intel.com>
Signed-off-by: Xiaotian Chen <xiaotian.chen@intel.com>

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -108,6 +108,13 @@ else
     do_that();
 ```
 
+### Error message
+Avoid using "Uable to xxxx", use "Failed to xxx" instead for logging failed function calls:
+``` c
+// CNE_ERR_GOTO(out, "Unable to init CNE\n");
```
[the remark this answers is not in the document]
Comment (haoruan) [reply CONTRIBUTING.md:114]: The commented code is just the old code, here I use this code as an example.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -108,6 +108,13 @@ else
     do_that();
 ```
 
+### Error message
+Avoid using "Uable to xxxx", use "Failed to xxx" instead for logging failed function calls:
```
[the remark this answers is not in the document]
Comment (haoruan) [reply CONTRIBUTING.md]: Thanks, updated

File: CONTRIBUTING.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (KeithWiles) [reply CONTRIBUTING.md:114]: Maybe we should show the example usage.
``` c
  CNE_ERR_GOTO(out, "Unable to init CNE\n");  // Do NOT use 'unable to' phase here

  CNE_ERR_GOTO(out, "Failed to init CNE\n");  // Use 'failed to' instead.
```

**Comment by KeithWiles** [comment]
The clang-format failed `} * req;` should be `} *req;`

File: examples/cnet-quic/quic-cli.c [hunk 3]
```diff
@@ -114,7 +114,7 @@ static struct {
 struct {
     const char *path;
     int to_file;
-} *reqs;
+} * reqs;
```
Comment (KeithWiles) [inline examples/cnet-quic/quic-cli.c:117]: Remove the space between '* reqs' to get clang-format to pass. This should have been fixed with the per-commit hook running clang-format.

**Review by KeithWiles — CHANGES_REQUESTED** [review]
A few minor requests for change.

**Review by KeithWiles — APPROVED** [review]
LGTM

**Comment by haoruan** [comment]
Thanks for updating the PR @maryamtahhan

**Pull request merged**